libethcore/EthashGPUMiner.cpp

```diff
@@ -85,14 +91,33 @@ class EthashCLHook: public ethash_cl_miner::search_hook
 //		std::cerr << "Searched " << _count << " from " << _startNonce << std::endl;
 		m_owner->accumulateHashes(_count);
 		m_last = _startNonce + _count;
-		if (m_abort || m_owner->shouldStop())
-			return (m_aborted = true);
-		return false;
+		//if (m_abort || m_owner->shouldStop())
+		//	return (m_aborted = true);
+		//return false;
+		// TODO: still need a way to stop the GPU thread
```

Comment:
@pilsy I don't quite understand what you're describing, but I also have limited knowledge of openCL kernels. I just posted an [explanation](https://github.com/ethereum-mining/ethminer/pull/4#issuecomment-312116350) of the optimization in this PR. Perhaps you could expand a bit on your scheme, in the convo thread?

The only other code I haven't yet pushed is some profiling lines. I'm hoping to finally push the profiling code over the next few days, and double check the improvement.